docs/DefaultBundler.md

```diff
@@ -0,0 +1,376 @@
+
+For Example, integration/html-js-dedup/index.html
+
+```html
+<!DOCTYPE html>
+<script type="module" src="component-1.js"></script>
+<script type="module" src="component-2.js"></script>
+```
+
+![image info](./BundlerGraphs/steps/idealBundleGraph_htmldedup.png)
+
+![image info](./BundlerGraphs/steps/reachableRoots_htmldedup.png)
+
+What is ancestorAssets ? What about the output ?
+
+```// ancestorAssets
+index.html => {},
+component1 => {},
+component2 => {html-js-dedup/component-1.js}
+```
+
+Because component1 bundle loads first, there is no need to add `obj` to both siblings. `Index.html` is isolated and thus its availability is not propagated down to the children.
+
+```
+Entry Bundle =>     [index.html]
+Sibling Bundle 1 => [component1, obj.js]
+Sibling Bundle 2 => [component2]
+```
+
+## Step: Internalize Async Bundles
+
+Internalization is when some bundle requires an asset synchronously, but also asynchronously. This is redundant so we do not load the extra (async) bundle. We mark this in `bundle.internalizedAssetIds`, and an internalized asset is ultimately displayed as an orange edge.
+
+In previous iterations of the bundler, this step would have been costly, since to determine if an asset is available synchronously, we would've traversed the graph upwards in search of it. Now we simply consult `reachableRoots` and `ancestorAssets`, which tells us if the asset we want to load via asynchronous bundle is either available to us synchronously or via our ancestors.
+
+```js
+if (
+  reachableAssets[parentId].has(bundleRootId) ||
+  ancestorAssets[parentId]?.has(bundleRootId)
+) {
+  // internalize
+}
+```
+
+A bundle can be **deleted** in this step if we determine that all those who request this particular asynchronous bundle already have it available to them by some other means. We denote this with `canDelete`.
+
+## Step: Insert or Share
+
+Here, we finally place assets into bundles which require them. All of the previous work we've done assumes all synchronous assets will be available to the bundle that needs it, but we haven't actually placed those assets in yet. If an asset is “reachable” from many bundles, we can extract it into a shared bundle.
+
+You may think of reachable as all the bundles that still need an asset by some means. We filter it down to ensure we only place assets where they need to be. Any bundle that contains the asset in question in its ancestorAssets is filtered out, & entries are filtered out.
+
+### Reused Bundles
+
+There’s a special case here that is unique to the experimental bundler, which is reusing bundles. We noticed sometimes, if two or more bundles shared the whole contents of another bundle, reusing that bundle is as simple as drawing an edge, as opposed to creating a shared bundle that would essentially be a copy of another.
```

Comment:
not experimental anymore... there's a few other references to it in these docs as well